packages/node/src/soroban/api.soroban.ts

```diff
@@ -0,0 +1,142 @@
+    const { hostname, protocol, searchParams } = new URL(endpoint);
+
+    const protocolStr = protocol.replace(':', '');
+
+    logger.info(`Api host: ${hostname}, method: ${protocolStr}`);
+    if (protocolStr === 'https' || protocolStr === 'http') {
+      const options: Server.Options = {
+        //headers: {
+        //  'User-Agent': `Subquery-Node ${packageVersion}`,
+        //},
+        allowHttp: protocolStr === 'http',
+      };
+      //searchParams.forEach((value, name, searchParams) => {
+      //  (connection.headers as any)[name] = value;
+      //});
+      this.client = new SorobanServer(endpoint, options);
+    } else if (protocolStr === 'ws' || protocolStr === 'wss') {
+      this.client = new SorobanServer(this.endpoint);
+    } else {
+      throw new Error(`Unsupported protocol: ${protocol}`);
+    }
+  }
+
+  async init(): Promise<void> {
+    const network = await this.client.getNetwork();
+    this.chainId = network.passphrase;
+  }
+
+  async getFinalizedBlock(): Promise<SorobanRpc.GetLatestLedgerResponse> {
+    return this.client.getLatestLedger();
+  }
+
+  async getFinalizedBlockHeight(): Promise<number> {
+    return (await this.getFinalizedBlock()).sequence;
+  }
+
+  async getBestBlockHeight(): Promise<number> {
+    return (await this.client.getLatestLedger()).sequence;
+  }
+
+  getRuntimeChain(): string {
+    return this.name;
+  }
+
+  getChainId(): string {
+    return this.chainId;
+  }
+
+  getGenesisHash(): string {
+    return this.getChainId();
+  }
+
+  getSpecName(): string {
+    return 'Soroban';
+  }
+
+  async getEvents(height: number): Promise<SorobanRpc.GetEventsResponse> {
+    return this.client.getEvents({ startLedger: height, filters: [] });
+  }
+
+  async fetchBlock(
+    blockNumber: number,
+    includeTx?: boolean,
+  ): Promise<SorobanBlockWrapped> {
+    try {
+      const events = await this.getEvents(blockNumber);
+
+      const ret = new SorobanBlockWrapped(events.events ?? [], {
+        height: blockNumber,
+        hash: blockNumber.toString(),
+      } as SorobanBlock);
+
+      this.eventEmitter.emit('fetchBlock');
+      return ret;
+    } catch (e) {
+      throw this.handleError(e);
+    }
+  }
+
+  async fetchBlocks(bufferBlocks: number[]): Promise<SorobanBlockWrapper[]> {
+    return Promise.all(
+      bufferBlocks.map(async (num) => this.fetchBlock(num, true)),
+    );
+  }
+
+  get api(): Server {
+    return this.client;
+  }
+
+  getSafeApi(blockHeight: number): SafeSorobanProvider {
+    return new SafeSorobanProvider(this.client, blockHeight);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async connect(): Promise<void> {
+    logger.error('Soroban API connect is not implemented');
+    throw new Error('Not implemented');
+  }
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async disconnect(): Promise<void> {
+    logger.error('Soroban API disconnect is not implemented');
+    throw new Error('Not implemented');
```

Comment:
It looks like we support WS from the constructor. How is connection managed for that?

Reply:
soroban client does not support ws yet.